.github/workflows/check-whitespace.yml

```diff
@@ -0,0 +1,30 @@
+name: check-whitespace
+
+# Get the repo with the last 30 commits.
+# Find the ref for the last merge and use it to limit the
+# log check range.
+
+on:
+  push:
+    branches-ignore: [ master, main ]
+
+jobs:
+  check-whitespace:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 30
+    - name: git log --check
+      run: |
+        m=$(git rev-list -1 --merges HEAD)
+        if [ -n "$m" ];
+        then
+          r="$m..HEAD"
+          echo "Setting range $r"
+        fi
+        echo ""
+        echo "Run git log command..."
+        echo ""
+        git log --oneline --check $r
```

Comment:
How about using `git log --oneline --check ${{github.base_ref}}..${{github.head_ref}}` (see [here](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context) for context)?

Reply:
Unfortunately those properties are only available on a pull request.  I did not find anything similar available on the github api.  Since git uses merges and gitgitgadget rejects merges in a pull request this seemed like a reasonable way to limit the check.  Ideally the `depth:` on the checkout would limit the commits but there seems to be chicken and egg problem.  Need to run `rev-list --count` (or something) to get the count before getting the repo.  If only github allowed running a git command against the remote.  Of course, I could have easily missed something obvious while I was wandering around in the weeds. 🙄

Reply:
But if we're _already_ targeting GitGitGadget users, we can limit this action to `on: pull_request`, no?

Reply:
When I asked about this in gitgitgadget issue # 169 (link above) I did suggest it would be for a push request to get the check done early.  If you would like it to be for a pull request, I will change it and use the available properties.  If it is for a pull request, would you like the error output added to the PR as a comment?  That would make it easier for the user.

Reply:
> If it is for a pull request, would you like the error output added to the PR as a comment?

That would be nice, but keep in mind that the GitHub action is run also on other forks than just git/git and gitgitgadget/git. So that PR comment would need to be added, say, in a [GitHub Script](https://github.com/marketplace/actions/github-script) task.

Reply:
> keep in mind that the GitHub action is run also on other forks than just git/git and gitgitgadget/git. So that PR comment would need to be added, say, in a [GitHub Script](https://github.com/marketplace/actions/github-script) task.

Those forks would need to enable actions first.  The action will now add a comment and also show the errors in the action log, so the user can see it whichever way they prefer.

fyi, `${{github.base_ref}}..${{github.head_ref}}` is not used because `github.base.ref` is the branch and not present in the minimally cloned repo.  Using the number of commits works fine.  An extra commit is part of the checkout so git does not think everything was added in the first commit and included in the check.  TIL while testing.